### ISSUE TITLE
Add feature to run specific test files.

### ISSUE THREAD

**Issue by codetalker7** [JuliaLang/Pkg.jl#3354]
For instance, frameworks like `jest` allow the user to run tests [which follow a particular pattern](https://jestjs.io/docs/cli#running-from-the-command-line). Currently, in Julia, only the following two methods exist for testing: 

- Use the `Test.jl` package and run tests manually in a Julia session.
- Test a package from `pkg` mode: 
```julia
(packageName) pkg> test
```

There is no systematic way in `Pkg.jl` to run specific test files. Packages like [TestSetExtensions.jl](https://github.com/ssfrr/TestSetExtensions.jl) exist, but I think this should be part of Julia's core functionality.

I can think of two possible solutions: 

- An interface to mention particular files to test; for example, suppose the `test/` directory contains files `a.jl` and `b.jl`. To run the tests in only these files, one might do: 
```julia
(packageName) pkg> test a b
```

- To allow running tests whose names match against a pattern. For example, if I have a `@testset`
```julia
@testset "myfunction()" begin
    # some tests
end
```
Then I should be able to do

```julia
(packageName) pkg> test "myfunction()"
```

This will make testing much easier and convenient.

**Comment by fredrikekre** [comment]
The Pkg REPL can perhaps pass arguments after the package and pass that to `test_args` which `Pkg.test` support already.
```
pkg> test Example -- a b
```

**Comment by codetalker7** [comment]
> 

@fredrikekre thanks for this. However, `Pkg.jl`'s documentation doesn't explain much about `test_args`; will it be a good idea to add a few examples to the API reference for this?

Also, regarding your idea: does the Pkg REPL already have this feature, or is this yet to be implemented?

**Comment by fredrikekre** [comment]
No this doesn't work right now, was just an idea for what can be done from Pkg.jl's side. Filtering based on those arguments would then be done by the test framework you are using.

**Comment by codetalker7** [comment]
> No this doesn't work right now, was just an idea for what can be done from Pkg.jl's side. Filtering based on those arguments would then be done by the test framework you are using.

Yes, I think that would be a good idea. If you can provide me a starting point for this, I can work on this feature and make a PR.

**Comment by fredrikekre** [comment]
Probably would start here: https://github.com/JuliaLang/Pkg.jl/blob/957b55a896d5cb496da134ea7bf3ee70de07ef2a/src/REPLMode/command_declarations.jl#L9 by defining a new function that splits on `--` and then calls `parse_package` on the left and some other parsing of the right part, then thread that into the API call somewhere down the line
